SolutionsAndTests

```diff
@@ -0,0 +1,258 @@
+function problem1(numbers)
+{
+	// safety checks, test for param null or not an array
+	if(	numbers == null)
+	{
+		return 0;
+	}
+	if(!(Object.prototype.toString.call( numbers ) === '[object Array]' ))
+	{
+		return 0;
+	}
+	// loop every element
+	var sum = 0;
+	for(i = 0; i < numbers.length; i++)
+	{
+		if(numbers[i] % 2 == 1 || numbers[i] % 2 == -1)
```

Comment:
Problem is:'Given an integer array of size N, compute the sum of all even numbers in this array'

This if is only true for odd numbers instead of even numbers.
Even numbers are ````numbers[i]%2 == 0